core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -470,6 +470,12 @@ default void setColumnType(ConnectorSession session, ConnectorTableHandle tableH
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting column types");
     }
 
+    @Experimental(eta = "2023-09-01")
+    default void setFieldType(ConnectorSession session, ConnectorTableHandle tableHandle, List<String> fieldPath, Type type)
```

Comment:
Document `fieldPath`. Especially case-sensitivity of the path components should be documented.

Reply:
Can `fieldPath` be empty or single-element?